src/main/frontend/db/query_react.cljs

```diff
@@ -112,17 +112,20 @@
          f)) query)))
 
 (defn react-query
-  [repo {:keys [query inputs] :as query'} query-opts]
+  [repo {:keys [query inputs rules] :as query'} query-opts]
   (let [pprint (if config/dev? (fn [_] nil) debug/pprint)]
     (pprint "================")
     (pprint "Use the following to debug your datalog queries:")
     (pprint query')
     (try
       (let [query (resolve-query query)
-            inputs (map resolve-input inputs)
+            resolved-inputs (mapv resolve-input inputs)
+            inputs (cond-> resolved-inputs
+                           rules
+                           (conj rules))
             repo (or repo (state/get-current-repo))
             k [:custom query']]
-        (pprint "inputs (post-resolution):" inputs)
+        (pprint "inputs (post-resolution):" resolved-inputs)
         (pprint "query-opts:" query-opts)
         (apply react/q repo k query-opts query inputs))
```

Comment:
@tiensonqin Rules get passed as an additional input. I think this works worth the `react/q` caching but wanted to point it out in case you have a concern

Reply:
This should be ok. 
It'll be great if users can write queries like:
```
#+BEGIN_QUERY
{:title "All pages that have a :foo property "
 :query [:find (pull ?p [*])
         :where
         (has-page-property ?p :foo)]}
#+END_QUERY
```

Reply:
Yep! I'm working on porting these rules to advanced query